ngraph/core/include/openvino/core/rtti.hpp

```diff
@@ -11,24 +11,26 @@
 
 #define _OPENVINO_RTTI_WITH_TYPE(TYPE_NAME) _OPENVINO_RTTI_WITH_TYPE_VERSION(TYPE_NAME, "util")
 
-#define _OPENVINO_RTTI_WITH_TYPE_VERSION(TYPE_NAME, VERSION_NAME)                  \
-    static const ::ov::DiscreteTypeInfo& get_type_info_static() {                  \
-        static const ::ov::DiscreteTypeInfo type_info{TYPE_NAME, 0, VERSION_NAME}; \
-        return type_info;                                                          \
-    }                                                                              \
-    const ::ov::DiscreteTypeInfo& get_type_info() const override {                 \
-        return get_type_info_static();                                             \
+#define _OPENVINO_RTTI_WITH_TYPE_VERSION(TYPE_NAME, VERSION_NAME)            \
+    static const ::ov::DiscreteTypeInfo& get_type_info_static() {            \
+        static ::ov::DiscreteTypeInfo type_info{TYPE_NAME, 0, VERSION_NAME}; \
+        type_info.hash();                                                    \
+        return type_info;                                                    \
+    }                                                                        \
+    const ::ov::DiscreteTypeInfo& get_type_info() const override {           \
+        return get_type_info_static();                                       \
     }
 
 #define _OPENVINO_RTTI_WITH_TYPE_VERSION_PARENT(TYPE_NAME, VERSION_NAME, PARENT_CLASS) \
     _OPENVINO_RTTI_WITH_TYPE_VERSIONS_PARENT(TYPE_NAME, VERSION_NAME, PARENT_CLASS, 0)
 
 #define _OPENVINO_RTTI_WITH_TYPE_VERSIONS_PARENT(TYPE_NAME, VERSION_NAME, PARENT_CLASS, OLD_VERSION) \
     static const ::ov::DiscreteTypeInfo& get_type_info_static() {                                    \
-        static const ::ov::DiscreteTypeInfo type_info{TYPE_NAME,                                     \
-                                                      OLD_VERSION,                                   \
-                                                      VERSION_NAME,                                  \
-                                                      &PARENT_CLASS::get_type_info_static()};        \
+        static ::ov::DiscreteTypeInfo type_info{TYPE_NAME,                                           \
+                                                OLD_VERSION,                                         \
+                                                VERSION_NAME,                                        \
+                                                &PARENT_CLASS::get_type_info_static()};              \
+        type_info.hash();                                                                            \
```

Comment:
why do we need to call it here if it can be computed during first usage?

Reply:
in the first time this method can be called for the const object.

Reply:
What about moving it into the constructor as type info is immutable?

Reply:
If I remember right I had some issues if I remove `constexpr` annotation from the constructor.
So I will try to remove this annotation one more time.